src/components/application_manager/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -73,28 +73,20 @@ mobile_apis::AppHMIType::eType StringToAppHMIType(const std::string &str) {
 }
 
 std::string AppHMITypeToString(mobile_apis::AppHMIType::eType type) {
-  if (mobile_apis::AppHMIType::DEFAULT == type) {
-    return "DEFAULT";
-  } else if (mobile_apis::AppHMIType::COMMUNICATION == type) {
-    return "COMMUNICATION";
-  } else if (mobile_apis::AppHMIType::MEDIA == type) {
-    return "MEDIA";
-  } else if (mobile_apis::AppHMIType::MESSAGING == type) {
-    return "MESSAGING";
-  } else if (mobile_apis::AppHMIType::NAVIGATION == type) {
-    return "NAVIGATION";
-  } else if (mobile_apis::AppHMIType::INFORMATION == type) {
-    return "INFORMATION";
-  } else if (mobile_apis::AppHMIType::SOCIAL == type) {
-    return "SOCIAL";
-  } else if (mobile_apis::AppHMIType::BACKGROUND_PROCESS == type) {
-    return "BACKGROUND_PROCESS";
-  } else if (mobile_apis::AppHMIType::TESTING == type) {
-    return "TESTING";
-  } else if (mobile_apis::AppHMIType::SYSTEM == type) {
-    return "SYSTEM";
-  } else {
-    return NULL;
+  switch(type) {
+    case mobile_apis::AppHMIType::DEFAULT:            return "DEFAULT";
+    case mobile_apis::AppHMIType::COMMUNICATION:      return "COMMUNICATION";
+    case mobile_apis::AppHMIType::MEDIA:              return "MEDIA";
+    case mobile_apis::AppHMIType::MESSAGING:          return "MESSAGING";
+    case mobile_apis::AppHMIType::NAVIGATION:         return "NAVIGATION";
+    case mobile_apis::AppHMIType::INFORMATION:        return "INFORMATION";
+    case mobile_apis::AppHMIType::SOCIAL:             return "SOCIAL";
+    case mobile_apis::AppHMIType::BACKGROUND_PROCESS: return "BACKGROUND_PROCESS";
+    case mobile_apis::AppHMIType::TESTING:            return "TESTING";
+    case mobile_apis::AppHMIType::SYSTEM:             return "SYSTEM";
+    default:
+      NOTREACHED();
```

Comment:
@jacobkeeler NOTREACHED will be switched off in release mode, and you will just try to return NULL.
Compile will generate std::string(NULL) that will cause core crash,